src/arcaflow_plugin_sdk/schema.py

```diff
@@ -5439,6 +5438,17 @@ def serialize(self, data: Any, path: typing.Tuple[str] = tuple([])) -> Any:
 
     def _check(self, data: Any, path: typing.Tuple[str] = tuple([])):
         if isinstance(data, list):
+            if len(data) != 0:
+                list_type_base = type(data[0])
+                for i in range(len(data)):
+                    if type(data[i]) != list_type_base:
```

Comment:
In this case, you're not using the index, so it's better practice to just do `for item in data`, which will give you the items directly.